cmd/rpcdaemon/main.go

```diff
@@ -50,7 +52,22 @@ func main() {
 		defer db.Close()
 		defer engine.Close()
 
-		apiList := jsonrpc.APIList(db, backend, txPool, mining, ff, stateCache, blockReader, cfg, engine, logger, nil)
+		polygonSync, err := cmd.PersistentFlags().GetBool("polygon.sync")
+		if err != nil {
+			return err
+		}
+
+		var bridgeReader bridge.ReaderService
+		if polygonSync {
+			stateReceiverContractAddress := params.BorMainnetChainConfig.Bor.GetStateReceiverContract() // mainnet and amoy share the same address
+			bridgeReader, err = bridge.AssembleReader(ctx, cfg.DataDir, logger, stateReceiverContractAddress)
```

Comment:
I think this won't work for the remote rpcdaemon case.

When rpcdaemon is run remotely - it means that it is run on another machine.

Erigon is run on Machine A.
Rpcdaemon is run on Machine B.

Machine A has "erigon-datadir/bridge/mdbx.dat" available.
Machine B does not have it.

So any DB queries that the bridge reader has need to go via grpc. In other words we need RemoteBridgerReader implementation.

Have you test this case?

Reply:
Yes, in this case the `polygon.sync` flag does not need to be passed.

Reply:
I have updated the code so that `bridgeReader` is only used if `dataDir` and `polygon.sync` are both present.

Reply:
Why?
The task is to support remote rpcdaemon.

Reply:
The PR title says: "Integrate Astrid bridge with standalone and remote rpcdaemon"

Reply:
https://github.com/erigontech/erigon/blob/3ee834a405a5c53b5cd78317e2e2a80198197e1c/ethdb/privateapi/ethbackend.go#L291

It is handled by the machine running `erigon`, not `rpcdaemon`. The `rpcdaemon` process does not need to know if the `erigon` process is using `polygon.sync` or not since it is just querying for data.

Reply:
@shohamc1 Ok, I think I see where the confusion is coming from. You are right that `ETHBACKENDClient` handles this. 

However, currently `ETHBACKENDClient.BorEvent` is called by `RemoteBlockReader`.  What I think we should do is have `ETHBACKENDClient.BorEvent` be called by a `RemoteBridgeReader`. So when a separate rpcdaemon process is started locally on the same machine with  `--datadir` we use `bridge.Reader` but when we use a remote rpcdaemon on another machine we use `bridge.RemoteReader` (which calls ETHBACKENDClient under the hood).

The reason behind this is because the work we've been doing so far with Astrid has been to move BlockReader methods related to Bor outside of the BlockReader. This should include RemoteBlockReader too. Ie, when using Astrid standalone, we should not be using the RemoteBlockReader in the RPCs.